src/pages/[slug]/[ie]/index.tsx

```diff
@@ -246,16 +247,18 @@ const ObjectDetailPage: NextPage = () => {
 	useEffect(() => {
 		let backLink = `/${router.query.slug}`;
 		if (previousUrl) {
-			const subgroups = previousUrl?.match(/(?:[^/\n]|\/\/)+/gi);
-			const validBacklink =
-				subgroups?.length === 1 && subgroups[0]?.startsWith(router.query.slug as string);
+			const info = parseUrl(previousUrl);
+			const validBacklink = info?.url?.startsWith(('/' + router.query.slug) as string);
```

Comment:
Volgens mij gaan we hier problemen hebben als je doorklikt naar een andere `ie` (ook interessant of gerelateerd). Die starten ook met de naam van de space en dan kan je in een infinite loop geraken 🤔

Reply:
reverted the regex logic
kept the focus query param